src/frontend/komponenter/Fagsak/Vedtak/VedtakContainer.tsx

```diff
@@ -72,27 +70,49 @@ const VedtakContainer: React.FC<IProps> = ({ behandling, fagsak }) => {
         behandling.behandlingsårsakstype ===
             Behandlingårsak.REVURDERING_FEILUTBETALT_BELØP_HELT_ELLER_DELVIS_BORTFALT;
 
-    const { hentSjekkLikhetPerioder } = useSjekkLikhetPerioder(behandling.behandlingId);
+    const { erPerioderLike, hentSjekkLikhetPerioder } = useSjekkLikhetPerioder(
+        behandling.behandlingId
+    );
+    const { erPerioderSlåttSammen, hentErPerioderSlåttSammen } = useErPerioderSlåttSammen(
+        behandling.behandlingId
+    );
+    const { slåSammenPerioder, feilmelding } = useSlåSammenPerioder(
+        behandling.behandlingId,
+        !erPerioderSlåttSammen
+    );
+
+    const handleSlåSammenPerioder = async () => {
+        await slåSammenPerioder();
+        await hentSjekkLikhetPerioder();
+        await hentErPerioderSlåttSammen();
+        hentVedtaksbrevtekster();
+    };
 
     useEffect(() => {
-        hentSjekkLikhetPerioder();
-    }, [hentSjekkLikhetPerioder]);
+        const fetch = async () => {
+            await hentSjekkLikhetPerioder();
+            await hentErPerioderSlåttSammen();
```

Comment:
Det er problematisk at disse to kallene er "avhengig" av hverandre. Kommenterer mer i backend-koden sånn sett

Reply:
at `hentSjekkLikhetPerioder` har en sideeffekt er uheldig og vanskelig å forstå senere. Forslag til løsning i backend. Da kan også frontend gjøres litt enklere.